api/anilist/src/main/kotlin/com/imashnake/animite/api/anilist/FlowExt.kt

```diff
@@ -0,0 +1,19 @@
+package com.imashnake.animite.api.anilist
+
+import com.apollographql.apollo3.api.ApolloResponse
+import com.apollographql.apollo3.api.Operation
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.mapLatest
+import java.io.IOException
+
+@OptIn(ExperimentalCoroutinesApi::class)
+fun <T: Operation.Data, R> Flow<ApolloResponse<T>>.asResult(transform: (T) -> R): Flow<Result<R>> = this
+    .mapLatest {
+        // TODO Is there a better way to handle errors here?
+        it.data?.let { data ->
+            Result.success(transform(data))
```

Comment:
Are we sure that thrown exceptions in here gets caught by the `asResource()` transform? Especially if we're gonna be so strict about non-null data like in the files above with `it.page!!` and `it.media!!`

Only other thing I'd say we could do is integrate the safer checking of errors vs. non-null data like the *cough* removed code for it, as I would like a human readable error for no data, rather than an IOException with a potential empty string
```
if (response.data != null) {
  success<R?>(mapper(response.dataAssertNoErrors))
} else if (response.hasErrors()) {
  error(response.errors!!.joinToString { it.toString() })
} else {
  noDataError()
}
```

Reply:
This doesn't actually catch any thrown errors, it just propagates the first message from Apollo. Not sure if that's human-readable, or if there can ever be more than one, hence the TODO